src/Modal/Modal.test.jsx

```diff
@@ -0,0 +1,196 @@
+  describe('correct rendering', () => {
+    const buttons = [
+      <Button
+        display="Blue button!"
+        buttonType="primary"
+      />,
+      {
+        display: 'Red button!',
+        buttonType: 'danger',
+      },
+      <Button
+        display="Green button!"
+        buttonType="success"
+      />,
+    ];
+
+    it('renders default buttons', () => {
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+        />,
+      );
+      const modalTitle = wrapper.find('.modal-title');
+      const modalBody = wrapper.find('.modal-body');
+
+      expect(modalTitle.text()).toEqual(title);
+      expect(modalBody.text()).toEqual(body);
+      expect(wrapper.find('button')).toHaveLength(2);
+    });
+
+    it('renders custom buttons', () => {
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+          buttons={buttons}
+        />,
+      );
+      expect(wrapper.find('button')).toHaveLength(buttons.length + 2);
+    });
+  });
+
+  describe('props received correctly', () => {
+    it('component receives props', () => {
+      wrapper = mount(
+        <Modal
+          title={title}
+          body={body}
+        />,
+      );
+
+      modalOpen(false, wrapper);
+      wrapper.setProps({ open: true });
+      modalOpen(true, wrapper);
+    });
+    it('component receives props and ignores prop change', () => {
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+        />,
+      );
+
+      modalOpen(true, wrapper);
+      wrapper.setProps({ title: 'Changed modal title' });
+      modalOpen(true, wrapper);
+    });
+  });
+
+  describe('close functions properly', () => {
+    beforeEach(() => {
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+        />,
+      );
+    });
+
+    it('closes when x button pressed', () => {
+      modalOpen(true, wrapper);
+      wrapper.find('button').at(0).simulate('click');
+      modalOpen(false, wrapper);
+    });
+
+    it('closes when Close button pressed', () => {
+      modalOpen(true, wrapper);
+      wrapper.find('button').at(1).simulate('click');
+      modalOpen(false, wrapper);
+    });
+
+    it('closes when Escape key pressed', () => {
+      modalOpen(true, wrapper);
+      wrapper.find('button').at(0).simulate('keyDown', { key: 'Escape' });
+      modalOpen(false, wrapper);
+    });
+    it('calls callback function on close', () => {
+      const spy = jest.fn();
+
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+          onClose={spy}
+        />,
+      );
+
+      expect(spy).toHaveBeenCalledTimes(0);
+
+      // press X button
+      wrapper.find('button').at(0).simulate('click');
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+  });
+  describe('invalid keystrokes do nothing', () => {
+    beforeEach(() => {
+      wrapper = mount(
+        <Modal
+          {...defaultProps}
+        />,
+      );
+    });
+
+    it('does nothing on invalid keystroke q', () => {
```

Comment:
Lol I love this. I kind of feel like "invalid keystroke q" is going to become the go-to for coverage fixes.